src/store/reducers/SavedQueries.ts

```diff
@@ -0,0 +1,60 @@
+import {
+  QueryType,
+  SavedQueriesActions,
+  SavedQueriesActionTypes,
+  SavedQueriesState,
+  SavedQueriesStatuses,
+} from '../SavedQueriesTypes';
+
+const initialState: SavedQueriesState = {
+  isLoadingAllQueries: false,
+  queries: { [QueryType.file]: [], [QueryType.cohort]: [] },
+  queryIdToStatus: {},
+  errorFetchAllQueries: null,
+};
+
+export default (state = initialState, action: SavedQueriesActionTypes): SavedQueriesState => {
+  switch (action.type) {
+    case SavedQueriesActions.toggleFetchAllSavedQueries: {
+      return { ...state, isLoadingAllQueries: action.isLoadingAllQueries };
+    }
+    case SavedQueriesActions.addAllSavedQueries: {
+      return { ...state, queries: action.queries };
+    }
+    case SavedQueriesActions.failureFetchingAllSavedQueries: {
+      return { ...state, errorFetchAllQueries: action.error };
+    }
+    case SavedQueriesActions.requestDeleteSavedQuery: {
+      const queryId = action.queryId;
+
+      const queryIdToStatusShallowCopy = { ...state.queryIdToStatus };
+      queryIdToStatusShallowCopy[queryId] = SavedQueriesStatuses.deleting;
+
+      return { ...state, queryIdToStatus: queryIdToStatusShallowCopy };
```

Comment:
```
return { ...state, queryIdToStatus: { ...state.queryIdToStatus, [queryId]:  SavedQueriesStatuses.deleting} };
```
would also work i think

Reply:
Yes, it would but I have mixed feelings about it. It's a one-liner, but seems a bit more complex than it needs to. Maybe the other reviewers can give their take on it.

Reply:
I retract my last comment. It is indeed better.